test/test_release_build.py

```diff
@@ -0,0 +1,79 @@
+import datetime
+from unittest.mock import create_autospec
+
+from dateutil.tz import tzutc
+
+from exasol_script_languages_developer_sandbox.lib.aws_access.aws_access import AwsAccess
+from exasol_script_languages_developer_sandbox.lib.aws_access.stack_resource import StackResource
+from exasol_script_languages_developer_sandbox.lib.github_release_access import GithubReleaseAccess
+from exasol_script_languages_developer_sandbox.lib.release_build.run_release_build import run_start_release_build, \
+    run_start_test_release_build
+
+UPLOAD_URL = "https://uploads.github.com/repos/exasol/script-languages-developer-sandbox/releases/123/assets{?name,label}"
+BRANCH = "main"
+GITHUB_TOKEN = "gh_secret"
+
+DUMMY_RESOURCES = [
+    StackResource({'LogicalResourceId': 'CodeBuildLogGroup',
+     'PhysicalResourceId': '/aws/codebuild/log-id',
+     'ResourceType': 'AWS::Logs::LogGroup',
+     'LastUpdatedTimestamp': datetime.datetime(2022, 5, 4, 18, 39, 11, 935000, tzinfo=tzutc()),
+     'ResourceStatus': 'CREATE_COMPLETE', 'DriftInformation': {'StackResourceDriftStatus': 'NOT_CHECKED'}
+     }),
+    StackResource({'LogicalResourceId': 'CodeBuildRole',
+     'PhysicalResourceId': 'role-id',
+     'ResourceType': 'AWS::IAM::Role',
+     'LastUpdatedTimestamp': datetime.datetime(2022, 5, 4, 18, 39, 1, 806000, tzinfo=tzutc()),
+     'ResourceStatus': 'CREATE_COMPLETE', 'DriftInformation': {'StackResourceDriftStatus': 'NOT_CHECKED'}
+     }),
+    StackResource({'LogicalResourceId': 'developerSandboxReleaseCodeBuild',
+     'PhysicalResourceId': 'codebuild-id-123',
+     'ResourceType': 'AWS::CodeBuild::Project',
+     'LastUpdatedTimestamp': datetime.datetime(2022, 5, 4, 18, 39, 7, 850000, tzinfo=tzutc()),
+     'ResourceStatus': 'CREATE_COMPLETE', 'DriftInformation': {'StackResourceDriftStatus': 'NOT_CHECKED'}
+     })
+]
+
+
+def test_release_build(test_config):
+    """
+    Test that serialization and deserialization of KeyFileManager work!
+    """
+    aws_access_mock = create_autospec(AwsAccess)
+    aws_access_mock.get_all_stack_resources.return_value = DUMMY_RESOURCES
+    run_start_release_build(aws_access_mock, test_config, UPLOAD_URL, BRANCH, GITHUB_TOKEN)
+    expected_env_variable_overrides = [
+        {"name": "RELEASE_ID", "value": "123", "type": "PLAINTEXT"},
+        {"name": "ASSET_ID", "value": test_config.slc_version, "type": "PLAINTEXT"},
+        {"name": "GITHUB_TOKEN", "value": GITHUB_TOKEN, "type": "PLAINTEXT"}
+    ]
+
+    aws_access_mock. \
+        start_codebuild.assert_called_once_with("codebuild-id-123",
+                                                environment_variables_overrides=expected_env_variable_overrides,
+                                                branch=BRANCH)
+
+
+def test_test_release_build(test_config):
+    """
+    Test that serialization and deserialization of KeyFileManager work!
+    """
+    aws_access_mock = create_autospec(AwsAccess, spec_set=True)
```

Comment:
Maybe use union mock type hints and mock cast

Reply:
Done